app/javascript/controllers/date_controller.js

```diff
@@ -0,0 +1,67 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["preselected"]
+
+  connect() {
+    if (!this.isDisabled(this.preselectedTarget)) {
+      if (this.isDraft(this.preselectedTarget)) {
+        this.element.classList.add("bg-danger", "text-light")
+      } else {
+        this.element.classList.add("bg-primary", "text-light")
+      }
+    }
+
+    if (this.preselectedTarget.getAttribute("draft") === "true") {
+      this.element.classList.add("bg-warning")
+    }
+  }
+
+  toggleSelect(event) {
+    if (!this.isDisabled(event.target.firstElementChild)) {
+      if (this.isDraft(event.target.firstElementChild)) {
+        this.element.classList.remove("bg-danger")
+      }
+      event.target.classList.remove("bg-primary", "text-light")
+      event.target.firstElementChild.disabled = true
+    } else {
+      if (this.isDraft(event.target.firstElementChild)) {
+        event.target.classList.add("bg-danger", "text-light")
+      } else {
+        event.target.classList.add("bg-primary", "text-light")
+      }
+      event.target.firstElementChild.disabled = null
+    }
+  }
+
+  highlight(event) {
```

Comment:
`highlight` and `unhighlight` trigger on mouseover/mouseoff, to highlight the date the mouse is currently on.

Reply:
I'd actually suggest doing this in pure CSS with `:hover`.